code/datums/components/electrified.dm

```diff
@@ -0,0 +1,143 @@
+GLOBAL_LIST_INIT(unelectrifiable_machines, typecacheof(list(/obj/machinery/door/airlock/cult)))
+
+/datum/component/electrified
+	dupe_mode = COMPONENT_DUPE_UNIQUE_PASSARGS
+	var/electrification_state = MACHINE_NOT_ELECTRIFIED
+	var/duration
+	var/duration_timer
+
+/datum/component/electrified/Initialize(electrification_state, mob/user, duration = MACHINE_DEFAULT_ELECTRIFY_TIME)
+	log_electrification(src, user)
+
+	registersignals()
+
+	if(duration)
+		if(duration_timer && timeleft(duration_timer) >= duration)
+			return
+		duration_timer = addtimer(CALLBACK(src, .proc/end_electrification), duration, TIMER_STOPPABLE|TIMER_OVERRIDE)
+
+	RegisterSignal(src, COMSIG_ELECTRIFICATION_REMOVAL, .proc/end_electrification)

```

Comment:
This is what GetComponent is for, put removal behaviour in Destroy

Reply:
It's for the stuff where it toggles electrification, this signal is for that stuff to either <disable the electrification> and get a return value and if not then add electrification

Reply:
So should that use like
```
var/a = GetComponent(electrification)
if(a)
    qdel(a)
else
    AddComponent(electrification)
```
?